app/Http/Controllers/Twill/ExhibitionController.php

```diff
@@ -2,90 +2,74 @@
-            'sortKey' => 'aic_start_at',
-        ],
-    ];
-
-    protected $indexWith = ['medias'];
-
-    protected $formWith = ['revisions', 'siteTags'];
-
-    protected $defaultOrders;
-
-    protected $filters = [];
-
-    public function __construct(Application $app, Request $request)
+    public function setUpController(): void
     {
-        parent::__construct(...func_get_args());
-
-        $filters = $this->getRequestFilters();
-
-        if (is_array($filters) && ($filters['search'] ?? false)) {
-            $this->defaultOrders = ['_score' => 'desc'];
-        } else {
-            $this->defaultOrders = ['aic_start_at' => 'desc'];
-        }
+        $this->disableBulkDelete();
+        $this->disableBulkEdit();
+        $this->disableCreate();
+        $this->disableDelete();
+        $this->disableEdit();
+        $this->disableRestore();
+        $this->eagerLoadFormRelations(['revisions', 'siteTags']);
+        // I believe this was meant to load local images (as opposed to from the
+        // api), but this no longer seems to work.
+        $this->eagerLoadListingRelations(['medias']);
+        $this->enableAugmentedModel();
+        // The images don't always render, but this was also an issue before
+        // upgrading.
+        $this->enableShowImage();
+        $this->setModuleName('exhibitions');
+        $this->setPreviewView('site.exhibitionDetail');
     }
 
-    /**
-     * Use the default order scope from Twill.
-     * Added this as an exception on exhibitions because it's the only API listing that
-     * sorting has been implemented.
-     *
-     * @see App\Models\Api\Exhibition::scopeOrderBy
-     */
-    protected function orderScope(): array
+    protected function additionalIndexTableColumns(): TableColumns
     {
-        return ModuleController::orderScope();
+        $columns = TableColumns::make();
+        $columns->add(
+            Text::make()
+                ->title('Dates')
+                ->field('date')
+                ->sortable()
+                ->sortKey('aic_start_at')
+                ->sortByDefault(direction: 'desc')
+                ->customRender(self::renderDate(...))
+        );
+        return $columns;
     }
 
-    protected function indexData($request)
+    protected function renderDate(Exhibition $exhibition)
     {
-        return [];
+        $date = '';
+
+        // Strangely, we cannot use isset() or empty() here
+        $hasStart = $exhibition->aic_start_at !== null;
+        $hasEnd = $exhibition->aic_end_at !== null;
+
+        // These default gracefully to `now` if the attrs are empty
+        $start = $exhibition->asDateTime($exhibition->aic_start_at);
+        $end = $exhibition->asDateTime($exhibition->aic_end_at);
+
+        if ($hasStart) {
+            $date .= $start->format('m d Y');
+        }
+
+        if ($hasStart && $hasEnd) {
+            $date .= '–';
+        }
+
+        if ($hasEnd) {
+            $date .= $end->format('m d Y');
+        }
```

Comment:
Can you use `$exhibition->present()->date()` here, to avoid code duplication?

Reply:
I've updated the controller to use the existing presenter method.